logstash-core/lib/logstash/plugin.rb

```diff
@@ -92,7 +92,7 @@ def debug_info
   end
 
   def metric=(new_metric)
-    @metric = new_metric.namespace(@id)
+    @metric = new_metric
```

Comment:
This is actually good question, my expectations somehow match with yours, plugins should not be aware of namespaces, they should just get the right metrics object. But if you ask me about internal logstash location for this change I think it belongs to the plugin base class. Having this change in the delegators makes you have duplicated code and introduced potential issues with plugins when the metric object is passed without the "right" namespace. What do you think?